File: linux/RedBaronEdr/ecs_parser/convertKubeArmor.py
import json
import sys
from datetime import datetime

def convert_to_ecs(input_data):
    ecs_data = {
        "@timestamp": input_data.get("UpdatedTime", datetime.utcnow().isoformat() + "Z"),
        "host": {
            "name": input_data.get("HostName"),
            "id": str(input_data.get("UID", 0)),
            "os": {
                "platform": "linux"
            }
        },
        "process": {
            "name": input_data.get("ProcessName"),
            "pid": input_data.get("PID"),
            "ppid": input_data.get("HostPPID"),
            "args": input_data.get("Resource", "").split(),
            "executable": input_data.get("ProcessName"),
            "working_directory": input_data.get("Cwd"),
            "tty": {
                "device": input_data.get("TTY")
            },
            "parent": {
                "name": input_data.get("ParentProcessName"),
                "pid": input_data.get("HostPPID")
            }
        },
        "event": {
            "kind": "event",
            "category": ["process"],
            "type": ["start"],
            "action": input_data.get("Data"),
            "outcome": "success" if input_data.get("Result") == "Passed" else "failure"
        },
        "user": {
            "id": str(input_data.get("UID"))
        }
    }
    return ecs_data

def main(input_file):
    try:
        with open(input_file, 'r') as f:
            input_data = json.load(f)
        ecs_json = convert_to_ecs(input_data)
        print(json.dumps(ecs_json, indent=4))
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python convert_to_ecs.py <input_json_file>")
        sys.exit(1)
    
    input_file = sys.argv[1]
    main(input_file)

File: linux/RedBaronEdr/ecs_parser/convert_kubearmor.go
package ecs_parser

import (
	"encoding/json"
	"fmt"
	"strings"
)

func ConvertKubeArmorLog(inputData map[string]interface{}) (string, error) {
	// Create ECS data structure
	ecsData := map[string]interface{}{
		"@timestamp": inputData["UpdatedTime"],
		"host": map[string]interface{}{
			"name": inputData["HostName"],
			"id":   fmt.Sprintf("%v", inputData["UID"]),
			"os": map[string]interface{}{
				"platform": "linux",
				"type":     "linux",
			},
		},
		"process": map[string]interface{}{
			"name":              getProcessBaseName(fmt.Sprintf("%v", inputData["ProcessName"])),
			"pid":               inputData["PID"],
			"ppid":              inputData["PPID"],
			"args":              splitResource(fmt.Sprintf("%v", inputData["Resource"])),
			"executable":        inputData["ProcessName"],
			"working_directory": inputData["Cwd"],
			"tty": map[string]interface{}{
				"device": inputData["TTY"],
			},
			"parent": map[string]interface{}{
				"name":       getProcessBaseName(fmt.Sprintf("%v", inputData["ParentProcessName"])),
				"pid":        inputData["HostPPID"],
				"executable": inputData["ParentProcessName"],
			},
		},
		"event": map[string]interface{}{
			"kind":     "event",
			"category": []string{"process"},
			"type":     "start",
			"action": func() string {
				if fmt.Sprintf("%v", inputData["Data"]) == "syscall=SYS_EXECVE" {
					return "exec"
				}
				return fmt.Sprintf("%v", inputData["Data"])
			}(),
			"outcome": getOutcome(fmt.Sprintf("%v", inputData["Result"])),
		},
		"user": map[string]interface{}{
			"id": fmt.Sprintf("%v", inputData["UID"]),
		},
	}

	// Convert to JSON string
	jsonBytes, err := json.Marshal(ecsData)
	if err != nil {
		return "", fmt.Errorf("error marshaling ECS data: %v", err)
	}

	return string(jsonBytes), nil
}

// Helper function to split Resource string into args
func splitResource(resource string) []string {
	if resource == "" {
		return []string{}
	}
	return strings.Fields(resource)
}

// Helper function to convert Result to outcome
func getOutcome(result string) string {
	if result == "Passed" {
		return "success"
	}
	return "failure"
}

func getProcessBaseName(processName string) string {
	parts := strings.Split(processName, "/")
	return parts[len(parts)-1]
}
